src/test/java/com/vinyl/AddUserTest.java

```diff
@@ -0,0 +1,97 @@
+package com.vinyl;
+
+import com.vinyl.model.User;
+import com.vinyl.modelDTO.UserDTO;
+import com.vinyl.repository.UserRepository;
+import com.vinyl.service.PasswordEncoder;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import static org.mockito.ArgumentMatchers.any;
+
+public class AddUserTest extends BaseIntegration {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private BCryptPasswordEncoder originalPasswordEncoder;
+
+    private BCryptPasswordEncoder mockedPasswordEncoder;
+
+    UserDTO userDTO = new UserDTO();
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        setUserDTO(userDTO);
+        mockedPasswordEncoder = Mockito.mock(BCryptPasswordEncoder.class);
+        Mockito.when(mockedPasswordEncoder.encode(any())).thenReturn(userDTO.getPass());
+        passwordEncoder.setPasswordEncoder(mockedPasswordEncoder);
+    }
+
+    @Override
+    public void tearDown() {
+        passwordEncoder.setPasswordEncoder(originalPasswordEncoder);
+        super.tearDown();
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<User> response = trt.exchange("/users", HttpMethod.POST, new HttpEntity<>(userDTO), User.class);
```

Comment:
this line is duplicated here and in assertResponseWhenInvalid

Reply:
fixed